### PR TITLE
Add SecureDrop data flow diagram

### LINKED ISSUE #2370: Update Data Flow Diagram(s)
Our Data Flow Diagram is two years old and from the time of SecureDrop 0.3: 

dfd-0 3

We should make any updates since 0.3 and create more detailed DFDs for areas of the application with insufficient detail to aid in the enumerations of threats. For example, "source interface" and "journalist interface" are both just two small boxes in this diagram. For reference, see the Data Flow Diagrams section of the document [here](https://www.owasp.org/index.php/Application_Threat_Modeling).

### PR DESCRIPTION
**Pull request by emkll**
## Status

Ready for review 

## Description of Changes

Fixes #2370 .

- Moved Threat Model out of development section and into new Threat Model section.
- Added Data Flow diagram for SecureDrop in new Threat Model Section.
- Updated ToC, index, and broken links.

## Deployment

None

### If you made changes to the app code:

- [ ] Unit and functional tests pass on the development VM

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made changes to documentation:

- :white_check_mark:  Doc linting passed locally

### CODE REVIEW

File: docs/threat_model/dataflow.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Data Flow Diagram
+=================
+
+The following diagram captures all data flows to and from a SecureDrop deployment.
+
+|SecureDrop data flow diagram|
```
Comment (kushaldas) [inline docs/threat_model/dataflow.rst]: If we add the image in this style, it shows based on the width allowed in the theme, and default size is small enough that one may not be able to read the text in the image. If we use the directive below, it becomes a clickable (mouse over on the image gives an option to click) and by just clicking on the image one can open the full size image.

```
.. image:: ../diagrams/SecureDrop_DataFlow.png
   :scale: 100%
```

Comment (conorsch) [reply docs/threat_model/dataflow.rst]: Great catch, @kushaldas! Just tested both locally and @kushaldas's suggestion is indeed more intuitive. +1 for implementing.

**Comment by freddymartinez9** [comment]
I'm not sure `sneakernet` is a useful term here. Is there another term we can use?

**Comment by conorsch** [comment]
@freddymartinez9 The diff for this PR doesn't introduce any occurrences of "sneakernet", that's left over from the previous docs. The purpose of this PR is to close #2370 by updating the Data Flow Diagram.

File: docs/threat_model/dataflow.rst (hunk 1 above)
Comment (emkll) [reply docs/threat_model/dataflow.rst]: Agreed, this is way better. I will push the change, excellent suggestion @kushaldas !

**Comment by emkll** [comment]
Pushed changes addressing comments by @freddymartinez9 and @kushaldas

**Comment by conorsch** [comment]
Looks great. Now waiting for CI to finish post-revision. For future reference, you can prefix your feature branch with `docs-` and it'll skip the CI staging test suite—documented here: https://docs.securedrop.org/en/stable/development/documentation_guidelines.html#testing-documentation-changes

**Pull request merged**